src/main/java/org/webshar/hrms/service/LeaveApplicationService.java

```diff
@@ -164,35 +162,23 @@ public EmployeeLeaveApplicationResponse updateLeaveApplication(
     return leaveApplicationResponseBuilder.buildFromResult(updatedLeaveApplication);
   }
 
-  private int getNoOfDaysLeaveApplied(final Date startDate, final Date endDate) {
-    DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
-    LocalDate ldStartDate = LocalDate
-        .parse(new SimpleDateFormat(DATE_FORMAT).format(startDate), dateTimeFormatter);
-    LocalDate ldEndDate = LocalDate
-        .parse(new SimpleDateFormat(DATE_FORMAT).format(endDate), dateTimeFormatter);
-    return (int) ChronoUnit.DAYS.between(ldStartDate, ldEndDate) + 1;
+  private int getNoOfDaysLeaveApplied(final LocalDate startDate, final LocalDate endDate)
+  {
+    return (int) ChronoUnit.DAYS.between(startDate, endDate) + 1;
   }
 
   private void isStartAndEndDateAreOverlappingWithExistingRecordForGivenEmployeeAndLeaveTypeId(
-      final Long employeeId, final Long leaveTypeId, final Date startDate, final Date endDate)
-      throws EntityAlreadyExistsException {
```

Comment:
Why employee id is removed here. This verification should also include employee check parameter.